frontend/packages/console-app/src/components/quick-starts/QuickStartController.tsx

```diff
@@ -1,58 +1,32 @@
 import * as React from 'react';
-import { Dispatch, connect } from 'react-redux';
-import { RootState } from '@console/internal/redux';
-import * as QuickStartActions from '../../redux/actions/quick-start-actions';
-import { getActiveQuickStartState } from '../../redux/reducers/quick-start-reducer';
-import {
-  QuickStart,
-  QuickStartStatus,
-  QuickStartTaskStatus,
-  QuickStartState,
-} from './utils/quick-start-types';
+import { QuickStart, QuickStartStatus, QuickStartTaskStatus } from './utils/quick-start-types';
 import QuickStartContent from './controller/QuickStartContent';
 import QuickStartFooter from './controller/QuickStartFooter';
+import { QuickStartContext, QuickStartContextValues } from './utils/quick-start-context';
 
-type OwnProps = {
+type QuickStartControllerProps = {
   quickStart: QuickStart;
 };
 
-type StateProps = {
-  activeQuickStartState: QuickStartState;
-};
-
-type DispatchProps = {
-  setActiveQuickStart: (quickStartId: string) => void;
-  setQuickStartStatus: (quickStartId: string, status: QuickStartStatus) => void;
-  setQuickStartTaskNumber: (quickStartId: string, taskNumber: number) => void;
-  setQuickStartTaskStatus: (taskStatus: QuickStartTaskStatus) => void;
-};
-
-type MergeProps = {
-  status?: QuickStartStatus;
-  taskNumber?: number;
-  taskStatus?: QuickStartTaskStatus;
-  allTaskStatuses?: QuickStartTaskStatus[];
-};
-
-type QuickStartControllerProps = OwnProps & DispatchProps & MergeProps;
-
-const QuickStartController: React.FC<QuickStartControllerProps> = ({
-  quickStart,
-  status,
-  taskNumber,
-  taskStatus,
-  allTaskStatuses,
-  setActiveQuickStart,
-  setQuickStartStatus,
-  setQuickStartTaskNumber,
-  setQuickStartTaskStatus,
-}) => {
+const QuickStartController: React.FC<QuickStartControllerProps> = ({ quickStart }) => {
   const {
     metadata: { name },
     spec: { tasks },
   } = quickStart;
   const totalTasks = tasks?.length;
-
+  const {
+    activeQuickStartState,
+    setActiveQuickStart,
+    setQuickStartStatus,
+    setQuickStartTaskNumber,
+    setQuickStartTaskStatus,
+  } = React.useContext<QuickStartContextValues>(QuickStartContext);
+  const status = activeQuickStartState?.status as QuickStartStatus;
+  const taskNumber = activeQuickStartState?.taskNumber as number;
+  const allTaskStatuses = quickStart.spec.tasks.map(
```

Comment:
nit: we have `tasks` destructured above , can use it and will tasks always be there or a list if not good to have check